TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/EditorTests/PropertyTests/BoolPropertyTests.cs

```diff
@@ -0,0 +1,163 @@
+using System.Collections;
+using UnityEngine;
+using UnityEngine.TestTools;
+using NUnit.Framework;
+
+namespace UnityEditor.ShaderFoundry.UnitTests
+{
+    class BoolPropertyTests : BlockTestRenderer
+    {
+        BoolPropertyBlockBuilder BuildBoolWithoutNameOverrides(string defaultValue)
+        {
+            var propBuilder = new BoolPropertyBlockBuilder
+            {
+                PropertyAttribute = new PropertyAttributeData { DefaultValue = defaultValue },
+                FieldName = "BoolField",
+            };
+            return propBuilder;
+        }
+
+        BoolPropertyBlockBuilder BuildBoolWithNameOverrides(string defaultValue)
+        {
+            var propBuilder = new BoolPropertyBlockBuilder
+            {
+                PropertyAttribute = new PropertyAttributeData
+                {
+                    UniformName = "_Value",
+                    DisplayName = "Value",
+                    DefaultValue = defaultValue
+                },
+                FieldName = "BoolField",
+            };
+            return propBuilder;
+        }
+
+        [UnityTest]
+        public IEnumerator BoolProperty_DefaultPropertyValueUsed_IsExpectedColor()
+        {
+            var expectedColor = new Color(1, 0, 0, 0);
+            var container = new ShaderFoundry.ShaderContainer();
+
+            var propBuilder = BuildBoolWithNameOverrides("1");
```

Comment:
That's being checked inside of the block. The block is effectively doing:
```
if(input.value == 0)
  return trueColor;
else
  return falseColor;
```
I had just chosen to do colors instead of the integer value because it was more obvious when testing.

Reply:
Looks like bool block builder is just passing through the 0/1 value as the x component of the color directly, no?
builder.AddLine($"outputs.BaseColor = float3(inputs.{FieldName}, 0, 0);");

Should we test BuildBoolWithNameOverrides("0"); returns Color(0,0,0,0); ?

Reply:
Oh, I think I accidentally looked at BoolKeyword when I was double checking that. I'll add the 0 case (there's plenty more tests still needed, but I'd like to do a 2nd pass to add more once this gets in since it's blocking more work)